### PR TITLE
Blob HUD

### PR DESCRIPTION
**Pull request by DeityLink**
blob_hud

http://i.4cdn.org/vg/1462574364714.webm
- Points bar on the left, Health bar on the right. The health bar start flashing when it gets low.
- Buttons that require more points than you currently have appear grayscale.
- You can always jump back to your core using the small button at the top left. The first 24 special blobs that you've created (not counting the ones that got destroyed) also appear at the top, and you can quickly teleport from one to another by clicking them.

### CODE REVIEW

**Comment by PJB3005** [comment]
No.

You better modularize the HUD system in some way, adding to the pile of hardcoded garbage that is current HUD code is unacceptable.

**Comment by DeityLink** [comment]
@PJB3005 what are you talking about? I'm using the system we created for custom HUDs when I made the Adminbus, which is already pretty modular. Just how modular do you want it to be?

**Comment by DeityLink** [comment]
Alright as discussed with @PJB3005 I've regrouped all the Blob HUD buttons under the same obj/screen subtype. And I took this occasion to do the same for the Adminbus HUD buttons.

**Comment by PJB3005** [comment]
Alright, preferably we'd have a better HUD system but at the moment this is better I guess.

**Comment by d3athrow** [comment]
:+1:

**Comment by DeityLink** [comment]
That commit is for you @NigglyWiggly

**Comment by ChangelingRain** [comment]
> the bars are pillow shaded and don't even have numbers

wew

**Comment by DeityLink** [comment]
Of all the things that you can bitch about being pillow-shaded....bars, really? How would you shade bars? I'm not saying there isn't a better way to do it, I'm just saying you're aiming at a low-hanging fruit.

**Comment by DeityLink** [comment]
hnnng

re-added the numbers. oriented the shading on the gauges a bit.

**Comment by ihadtoregisterforthis** [comment]
Tubular

File: code/_onclick/hud/_defines.dm [hunk 1]
```diff
@@ -194,3 +194,17 @@
 #define ui_adminbus_tdarena "2:4,13:26"
 #define ui_adminbus_tdgreen "3:6,13:26"
 #define ui_adminbus_tdobs "2:4,14:28"
+
+//Blob HUD
+#define ui_blob_bg "1:0,1:0"
```
Comment (clusterfack) [inline code/_onclick/hud/_defines.dm]: Both these and the adminbus UIs need to use the system used above in the rest of the defines here.

Byond defines the corners of your screen as NORTH, SOUTH, EAST, WEST in the screen locs. By defining with NORTH,SOUTH,EAST,WEST your hud will automatically scale with the size of your screen.

So to align somethign with the bottom corner you make the screen loc "EAST,SOUTH", this is compatible with any other location through using EAST+x.

**Pull request merged**